### ISSUE TITLE
CPU shows 100% in Open Game Panel (Now 0% in Docker With PDS)

### ISSUE THREAD

**Issue by amsaal** [damentz/liquorix-package#59]
I am using the latest version 5.12.0-8.1 but when i go to game-panel , the CPU usage is 100% where as in SSH it shows way lower . so i am not sure what went wrong in this version the previous one was okay. i had to use some other custom kernel to show the correct load on game-panel status dashboard

can you please see why it shows 100% load on both virtual cores but on SSH actually there is no such load. 

Thanks.

**Comment by damentz** [comment]
I noticed this too, I think it's a regression from 5.12.8 and these commits:

https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.12.8&id=e325dc7acbdb4140dc7268caac1fa437e86edc51
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.12.8&id=a4367bbe866beba3a4ba8e448181832ab21f1885
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.12.8&id=a7fcb65fa1cf2a6da7264f4f61a309406f33cb9a

The CPU usage the kernel is reporting is purely cosmetic and doesn't mean that the CPU is running full throttle.

EDIT: Also note, this only happens under KVM virtualization.  I do not recommend running Liquorix for a server unless you _really_ know what you're doing.

**Comment by amsaal** [comment]
actually i like this kernel as the stock players complained 1 second lag in-game cs 1.6 goldsrc so after trying this kernel for 2 years my server is better .

the stock kernel is not optimized and does not have 1000 hz enabled by default so using yours i dont have to do all manually.

i use the lqx kernel in dedi and KVM VPS . 

if you can fix this CPU 100% show in Open Game Panel that would be much appreciated ,  may be some changes needed.

here is the screenshot it looks and my staff who are managing the server would like to see the normalize CPU usage so they can have less anxieties :)

https://snipboard.io/mrSnJa.jpg

Linux vps-2e710b7e.vps.ovh.net 5.12.0-8.2-liquorix-amd64 #1 ZEN SMP PREEMPT liquorix 5.12-8.1~buster (2021-05-31) x86_64 GNU/Linux
root@vps-2e710b7e:~#

Thanks and keep up the good work <3

UPDATE: i updated the kernel as i can see there was some changes to Linux 5.12.0-8.5-liquorix-amd64

but still it shows 100% CPU on Open Game Panel.

https://snipboard.io/oupVOS.jpg  [1-6-21]

**Comment by amsaal** [comment]
Hello,

stll same issue after the changes that was done yesterday, may be there is more needs to be done to revert back?

This is the one 5.12.0-8.5-liquorix-amd64 i am using shows 100 CPU in OGP

**Comment by damentz** [comment]
Yes, I did some testing myself, 5.12.0-8.5 is effectively 5.12.0-8.2.  Reverting those patches didn't help.

It seems more like this issue came up a version or two back and we're only now noticing.

**Comment by amsaal** [comment]
> Yes, I did some testing myself, 5.12.0-8.5 is effectively 5.12.0-8.2. Reverting those patches didn't help.
> 
> It seems more like this issue came up a version or two back and we're only now noticing.

this verion Linux 5.11.0-15.1-liquorix-amd64 (2021-04-16) was fine and i did not updated since months . it was just few days i started to update and notice this behaviour change sending 100 CPU load in OGP where as in console/ssh its all good .

**Comment by sapphonie** [comment]
I have a similar issue with [pterodactyl panel](https://github.com/pterodactyl/panel), showing 0% cpu usage as opposed to 100, on two seperate servers running Liquorix. Possibly related?

**Comment by damentz** [comment]
Yes that's related.  In-fact, pterodactyl, although affected, is actually very wrong anyways.  IO-wait actually means nothing is happening since we're waiting on IO. So showing 100% CPU when IO-wait is 100% tells you the opposite of what's happening.

There's two good node exporter dashboards for Grafana that accurately show when IO-wait is used over other types of CPU.  This requires prometheus as your datasource, of course.

As for when this will be fixed, I'm still waiting on @ckolivas to make an official port of MuQSS.  He might notice and fix the issue then.

**Comment by amsaal** [comment]
> Yes that's related. In-fact, pterodactyl, although affected, is actually very wrong anyways. IO-wait actually means nothing is happening since we're waiting on IO. So showing 100% CPU when IO-wait is 100% tells you the opposite of what's happening.
> 
> There's two good node exporter dashboards for Grafana that accurately show when IO-wait is used over other types of CPU. This requires prometheus as your datasource, of course.
> 
> As for when this will be fixed, I'm still waiting on @ckolivas to make an official port of MuQSS. He might notice and fix the issue then.

yes i would be happy if this gets fixed so i can roll out on my all VPS :)

keep up the good work

**Comment by damentz** [comment]
You are all probably aware that Con discontinued [1] MuQSS after noticing the colossal amount of work required to port to 5.14.  As such, Liquorix has switched to PDS, its spiritual successor developed by Alfred Chen: https://gitlab.com/alfredchen/linux-prjc

Due to the cleaner porting and better coding hygiene part of Alfred's work, issues like idle CPU being reported as IO-wait have been resolved.

Marking this issue as resolved.

[1] https://ck-hack.blogspot.com/2021/08/514-and-future-of-muqss-and-ck-once.html

**Comment by sapphonie** [comment]
Unfortunately this is still an issue in pterodactyl panel:

![image](https://user-images.githubusercontent.com/15315481/133941239-33741994-de7e-4de8-9625-de7b1ff43b38.png)


```
gen@server ~> sudo uname -a
Linux server 5.14.0-6.1-liquorix-amd64 #1 ZEN SMP PREEMPT liquorix 5.14-1.1~buster (2021-09-13) x86_64 GNU/Linux```

**Comment by damentz** [comment]
Hmm, that actually looks like a bug in Pterodactyl.  On my own server over KVM, the IO wait bug is fixed.

The gold standard for CPU usage displaying is `top` and `htop`.  If you get results different from that then you're misinterpreting system statistics.  In this case, Pterodactyl was already converting IO-wait to actual CPU usage incorrectly.  In the screenshots above, it's completely confused despite no other applications having issues.

I highly recommend you submit a bug report.  Seems there's a much larger issue in the underlying method that Pterodactyl reports CPU usage.

**Comment by sapphonie** [comment]
Thanks, will do

**Comment by sapphonie** [comment]
Yeah this is an issue with liq, not with ptero:
![image](https://user-images.githubusercontent.com/15315481/134079522-d8fdb973-a005-4c0b-ab5c-77ff98167fba.png)
Docker cpu usage with `docker stats` shows 0, which is... obviously incorrect

**Comment by sapphonie** [comment]
(Closed) issue with ptero: https://github.com/pterodactyl/panel/issues/3637

Should I open a new issue on this repo or just use this one?

**Comment by damentz** [comment]
@sapphonie we can use this one.  I suspect the issue may be that I enabled `CONFIG_CGROUP_ACCT` [1] as part of porting to 5.14 when I switched to PDS.  This CGROUP must be a stub on PDS.  Considering is mutually incompatible on MuQSS, there's probably a good reason for that considering PDS shares lineage with BFS/MuQSS.

I'll disable it and push out a new release.

[1] https://github.com/damentz/liquorix-package/commit/fb517be63376913e52ce237a530a775e26d60ee1#diff-02c4607130a689d3773e1389b975c9e437d3fadc6174cfddd35a814c35f47ed9R206

**Comment by damentz** [comment]
@sapphonie ok, I've already tried booting without `CGROUP_CONFIG_CPUACCT` with PDS and Docker still returns 0% CPU.  There's still a release on its way out with this change but it won't resolve this particular issue.

Opened an issue on `linux-prjc` directly, hopefully we can get some type of resolution: https://gitlab.com/alfredchen/linux-prjc/-/issues/39

**Comment by damentz** [comment]
@sapphonie ok, this bug is technically being worked on here: https://gitlab.com/alfredchen/linux-prjc/-/issues/35

Hopefully Alfred figures out what changes are needed and this issue will be fixed soon :crossed_fingers:

**Comment by damentz** [comment]
CPU accounting with Docker (and other similar things), should be resolved with 5.14-4 / 5.14.0-6.4.  Builds are underway, let me know if this fixes CPU usage in Pterodactyl.

**Comment by sapphonie** [comment]
This looks to be fixed. Thanks so much.
![image](https://user-images.githubusercontent.com/15315481/134403583-3c0b7137-7625-44ea-9139-ea113c8ff7a9.png)

**Comment by damentz** [comment]
Great!  Closing issue
